src/Backend/Modules/Groups/Actions/Delete.php

```diff
@@ -27,24 +28,26 @@ public function execute()
     {
         $this->id = $this->getParameter('id', 'int');
 
-        // group exists and id is not null?
-        if ($this->id !== null && BackendGroupsModel::exists($this->id)) {
-            parent::execute();
-
-            // get record
-            $this->record = BackendGroupsModel::get($this->id);
-
-            // delete group
-            BackendGroupsModel::delete($this->id);
-
-            // trigger event
-            BackendModel::triggerEvent($this->getModule(), 'after_delete', array('id' => $this->id));
-
-            // item was deleted, so redirect
-            $this->redirect(BackendModel::createURLForAction('Index') . '&report=deleted&var=' . urlencode($this->record['name']));
+        parent::execute();
+        $this->record = BackendGroupsModel::get($this->id);
+
+        if ($this->id !== null && !empty($this->record)) {
```

Comment:
if $this->id === null, you're record will always be empty, so no need to do both checks.